gateway/ln-gateway/src/rpc/rpc_server.rs

```diff
@@ -92,50 +90,8 @@ async fn auth_middleware(
     request: Request,
     next: Next,
 ) -> Result<impl IntoResponse, StatusCode> {
-    // These routes are not available unless the gateway's configuration is set.
-    let gateway_config = gateway
-        .clone_gateway_config()
-        .await
-        .ok_or(StatusCode::NOT_FOUND)?;
-    let gateway_hashed_password = gateway_config.hashed_password;
-    let password_salt = gateway_config.password_salt;
-    authenticate(gateway_hashed_password, password_salt, request, next).await
-}
-
-/// Middleware to authenticate an incoming request. Routes that are
-/// authenticated with this middleware are un-authenticated if the gateway has
-/// not yet been configured. After the gateway is configured, this middleware
-/// enforces that a Bearer token must be supplied in the Authorization header.
-async fn auth_after_config_middleware(
-    Extension(gateway): Extension<Arc<Gateway>>,
-    request: Request,
-    next: Next,
-) -> Result<impl IntoResponse, StatusCode> {
-    // If the gateway's config has not been set, allow the request to continue, so
-    // that the gateway can be configured
-    let gateway_config = gateway.clone_gateway_config().await;
-    if gateway_config.is_none() {
-        return Ok(next.run(request).await);
-    }
-
-    // Otherwise, validate that the Bearer token matches the gateway's hashed
-    // password
-    let gateway_config = gateway_config.expect("Already validated the gateway config is not none");
-    let gateway_hashed_password = gateway_config.hashed_password;
-    let password_salt = gateway_config.password_salt;
-    authenticate(gateway_hashed_password, password_salt, request, next).await
-}
-
-/// Validate that the Bearer token matches the gateway's hashed password
-async fn authenticate(
-    gateway_hashed_password: sha256::Hash,
-    password_salt: [u8; 16],
-    request: Request,
-    next: Next,
-) -> Result<axum::response::Response, StatusCode> {
     let token = extract_bearer_token(&request)?;
-    let hashed_password = hash_password(&token, password_salt);
-    if gateway_hashed_password == hashed_password {
+    if bcrypt::verify(token, &gateway.bcrypt_password_hash.to_string()).unwrap() {
```

Comment:
Is `unwrap` intentional here? If it is we should at least use `expect`

Reply:
Good catch! Done.